test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -321,14 +320,15 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 }
                 else
                 {
-                    using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                    using (var connection = new TestConnection(host.GetPort()))
                     {
-                        socket.Connect(new IPEndPoint(IPAddress.Loopback, host.GetPort()));
-                        socket.Send(Encoding.ASCII.GetBytes(
-                            "POST / HTTP/1.1\r\n" +
-                            "Transfer-Encoding: chunked\r\n" +
-                            "\r\n" +
-                            "wrong"));
+                        await connection.Send(
+                            "POST / HTTP/1.1",
+                            "Transfer-Encoding: chunked",
+                            "",
+                            "wrong");
+                        await connection.Receive(
```

Comment:
Yes. I had to change the test to use `TestServer`, and that required a few changes to `DummyApplication` too so I could pass in the mock context factory.